polars/polars-io/src/csv_core/utils.rs

```diff
@@ -52,6 +40,29 @@ pub(crate) fn get_file_chunks(
     offsets
 }
 
+pub(crate) fn get_reader_bytes<R: Read + MmapBytesReader>(reader: &mut R) -> Result<Vec<u8>> {
```

Comment:
Can this return a `Cow<[u8]>`?

The line `k(mmap[..].to_vec())` copies the whole file to memory before parsing, which has a bad impact on performance and memory.

Reply:
I tried to get this function to return Cow<[u8]> but it looks like it's not worth the effort:
https://stackoverflow.com/questions/68542207/return-resultcowu8-instead-of-resultvecu8/68547191#68547191
I think its best to just undo the function and repeat the code in infer_file_schema and as_df. Let me know if you think otherwise

Reply:
Yes, or an enum proposed as in the SO answer might also work. Seems a good solution to me.

Reply:
thanks, I've gone with the enum.